src/components/app/SingleStrategy/index.tsx

```diff
@@ -85,17 +80,27 @@ export const SingleStrategy = () => {
         vaultId,
         network = DEFAULT_NETWORK,
     } = useParams<ParamTypes>();
-
-    const [strategyData, setStrategyData] = useState<Strategy[]>([]);
-    const [isLoading, setIsLoading] = useState(true);
-    const [vault, setVault] = useState<Vault | undefined>();
-    const [isVaultLoading, setIsVaultLoading] = useState(true);
     const [isReportsLoading, setIsReportsLoading] = useState(true);
     const [error, setError] = useState<string | null>(null);
     const [value, setValue] = useState(0);
     const [warningFields, setWarningFields] = useState<string[] | null>(null);
     const [openSnackBar, setOpenSB] = useState(true);
     const strategyReportContext = useStrategyReportContext();
+    const {
+        data: vault,
+        error: fetchVaultError,
+        loading: fetchVaultLoading,
+    } = useVault(network, vaultId);
+    const {
+        data: strategy,
+        error: fetchStrategyError,
+        loading: fetchStrategyLoading,
+    } = useStrategy(network, vaultId, strategyId);
+    const {
+        data: strategyMetaData,
+        error: fetchstrategyMetaData,
+        loading: fetchstrategyMetaDataLoading,
```

Comment:
I wonder if we should load the hook in the StrategyDetail component so we can handle the loading flag separately? ie) Display `'Loading description...'` or something instead of `'Empty'`

Reply:
Yeah should work too, but maybe instead of passing down the hooks, we pass down the loading state?

Seems like a layout PR(next one), should include this.